lib/socket.c

```diff
@@ -2316,6 +2316,50 @@ uc_socket_error(uc_vm_t *vm, size_t nargs)
 		rv = ucv_stringbuf_finish(buf);
 	}
 
+	return rv;
+}
+
+/**
+ * Returns a string containing a description of the positive (`errno`) or
+ * negative (`EAI_*` constant) error code number given by the *code* argument.
+ *
+ * Returns `null` if the error code number is unknown.
+ *
+ * @function module:socket#strerror
+ *
+ * @param {number} code
+ * The error code.
+ *
+ * @returns {?string}
+ *
+ * @example
+ * // Should output 'Name or service not known'.
+ * print(socket.strerror(-2), '\n');
+ *
+ * // Should output 'No route to host'.
+ * print(socket.strerror(113), '\n');
+ */
+static uc_value_t *
+uc_socket_strerror(uc_vm_t *vm, size_t nargs)
+{
+	uc_value_t *codearg, *rv;
+	uc_stringbuf_t *buf;
+	int code;
+
+	args_get(vm, nargs, NULL,
+		"code", UC_INTEGER, false, &codearg);
+
+	code = ucv_to_integer(codearg);
+
+	buf = ucv_stringbuf_new();
```

Comment:
Since we're returning fixed strings here, the use of stringbuf is overkill. You can replace the `ucv_stringbuf_printf(buf, "%s", ...)` calls with a `ucv_string_new(...)`.